features/environment.py

```diff
@@ -358,23 +397,60 @@ def create_uat_ec2_image(context: Context, series: str) -> None:
             )
         )
         return
-    if context.config.build_pr:
-        raise RuntimeError("Don't yet support ec2 pro build_pr=1")
 
     # Launch pro image based on series marketplace lookup
     inst = launch_ec2(
         context, series=series, user_data=USERDATA_INSTALL_DAILY_PRO_UATOOLS
     )
-    print("Creating updated AWS PRO AMI from instance: {}".format(inst.id))
-    context.series_image_name[series] = context.config.ec2_api.snapshot(inst)
+    print("--- Creating updated AWS PRO AMI from instance: {}".format(inst.id))
+    context.series_image_name[series] = context.config.cloud_api.snapshot(inst)
     print(
-        "Created updated AWS PRO AMI: {}".format(
+        "--- Created updated AWS PRO AMI: {}".format(
             context.series_image_name[series]
         )
     )
 
 
-def create_uat_lxd_image(context: Context, series: str) -> None:
+def build_debs_from_dev_instance(context: Context, series: str) -> "List[str]":
+    """Create a development instance, instal build dependencies and build debs
+
+
+    Will stop the development instance after deb build succeeds.
+
+    :return: A list of paths to applicable deb files published.
+    """
+    time_suffix = datetime.datetime.now().strftime("%s%f")
+    if context.config.cloud_api:
```

Comment:
Maybe we could verify the machine type here before creating the ec2 instance

Reply:
Ahh right, I was hoping to generally use pycloudlib api for a cloud regardless of which cloud we are talking to EC2, versus Azure) for the instance launch. I expected our followup for Azure support to do something like this
```suggestion
    if context.config.cloud_api:
        context.config.cloud_api_launch_func(context, series=X1, user_data=X2)
```

For this branch I agree there is a bit of a mismatch, so we can change the check here to be more specific until we have multiple cloud support via pycloudlib.